int32 num_detections
---
string[] sorted_detections